ccmlib/common.py

```diff
@@ -971,6 +978,14 @@ def update_path_in_env(env, path_entry):
     return env
 
 
+def _maybe_set_use_jdk11_env(env, jvm_version, cassandra_version):
+    if jvm_version >= 11 and cassandra_version and '4.0' <= cassandra_version < '5.0':
+        env['CASSANDRA_USE_JDK11'] = 'true'
+    elif 'CASSANDRA_USE_JDK11' in env:
+        del env['CASSANDRA_USE_JDK11']
+    return env
```

Comment:
do we really want to introduce this ? given everything is so brittle and every change seems to be breaking _something_

otherwise i agree with the intent:  the caller may be specifying a binary version, thinking CASSANDRA_USE_JDK11 is never needed, but ccm can't download it and fallbacks to building the source of it…